### PR TITLE
feat(layout): enable odd/even header-footer support (w:evenAndOddHeaders)

### LINKED ISSUE #2803: feat: enable odd/even header-footer support (w:evenAndOddHeaders)
## What

Word documents with "Different Odd & Even Pages" enabled (`w:evenAndOddHeaders` in `word/settings.xml`) render the same header/footer on every page in SuperDoc. The even/odd variants are ignored.

## Root Cause

The layout engine hardcodes `alternateHeaders = false` at `packages/layout-engine/layout-engine/src/index.ts:1287`, preventing the paginator from selecting the correct header/footer variant per page. This causes incorrect margin calculations — the body content area doesn't account for different header heights on odd vs even pages.

The rest of the pipeline already supports this:
- **Import**: `w:evenAndOddHeaders` is parsed from `settings.xml` → `styles.alternateHeaders`
- **Layout Bridge**: `getHeaderFooterTypeForSection()` already handles even/odd variant selection
- **Rendering**: The provider pattern renders whichever variant is selected

## Expected Behavior

- Page 1 (odd): renders odd/default header
- Page 2 (even): renders even header  
- Page 3 (odd): renders odd/default header
- Margins adjust per page based on actual header/footer content height for each variant

## Spec Reference

- ECMA-376 §17.10 — Headers and Footers
- `w:evenAndOddHeaders` element in `word/settings.xml`

### PR DESCRIPTION
**Pull request by gpardhivvarma**
## Summary

Closes #2803

- Thread `alternateHeaders` from document settings (`w:evenAndOddHeaders`) through the layout engine so the paginator selects the correct header/footer variant per page
- Fix `getVariantTypeForPage` to use document page number (not section-relative) for even/odd selection, matching the rendering side (`headerFooterUtils.ts`)
- Add 4 unit tests covering variant selection, default behavior, and titlePg interaction

## Problem

Documents with "Different Odd & Even Pages" enabled render the same header/footer on every page. The layout engine hardcoded `alternateHeaders = false`, preventing correct margin calculations for odd vs even page headers.

## Changes

| File | Change |
|------|--------|
| `layout-engine/src/index.ts` | Add `alternateHeaders` to `LayoutOptions`, replace hardcoded `false`, fix `getVariantTypeForPage` to use document page number for even/odd |
| `layout-engine/src/index.test.ts` | 4 tests: even/odd selection, default false, omitted defaults, titlePg + alternateHeaders |
| `PresentationEditor.ts` | Thread `converter.pageStyles.alternateHeaders` into layout options |
| `types.ts` | Add `alternateHeaders` to `ResolvedLayoutOptions` |

## What was already working (no changes needed)

- Import: `w:evenAndOddHeaders` parsed from `settings.xml`
- Layout Bridge: `getHeaderFooterTypeForSection()` handles even/odd variant selection
- Rendering: provider pattern renders whichever variant is selected
- Export: `settings.xml` preserved as-is

## Test plan

- [x] `bun test` in layout-engine — 581/581 passing
- [x] 4 new alternateHeaders tests verify margin calculation per variant
- [x] Manual verification with a Word doc using "Different Odd & Even Pages"

### CODE REVIEW

**Comment by gpardhivvarma** [comment]
@caio-pizzol

File: packages/super-editor/src/editors/v1/core/presentation-editor/PresentationEditor.ts [hunk 1]
```diff
@@ -4238,10 +4238,12 @@ export class PresentationEditor extends EventEmitter {
...
       const blocksForLayout = semanticFootnoteBlocks.length > 0 ? [...blocks, ...semanticFootnoteBlocks] : blocks;
-      const layoutOptions =
-        !isSemanticFlow && footnotesLayoutInput
+      const layoutOptions = {
+        ...(!isSemanticFlow && footnotesLayoutInput
           ? { ...baseLayoutOptions, footnotes: footnotesLayoutInput }
-          : baseLayoutOptions;
+          : baseLayoutOptions),
+        alternateHeaders: Boolean((this.#editor as EditorWithConverter).converter?.pageStyles?.alternateHeaders),
+      };
```
Comment (caio-pizzol) [inline packages/super-editor/src/editors/v1/core/presentation-editor/PresentationEditor.ts]: `alternateHeaders` is always added but the type only declares it on the paginated branch of `ResolvedLayoutOptions`. no runtime issue, just a type mismatch.

```suggestion
      const layoutOptions = {
        ...(!isSemanticFlow && footnotesLayoutInput
          ? { ...baseLayoutOptions, footnotes: footnotesLayoutInput }
          : baseLayoutOptions),
        ...(isSemanticFlow
          ? {}
          : { alternateHeaders: Boolean((this.#editor as EditorWithConverter).converter?.pageStyles?.alternateHeaders) }),
      };
```

File: packages/layout-engine/layout-engine/src/index.test.ts [hunk 2]
```diff
@@ -5290,3 +5290,136 @@ describe('requirePageBoundary edge cases', () => {
     });
```
Comment (caio-pizzol) [inline packages/layout-engine/layout-engine/src/index.test.ts:5587]: all 4 tests here are single-section — the pre-PR code would pass them. worth adding:

1. multi-section — section 2 starting on an even page (the actual fix)
2. footer — top margin is covered, footer uses a separate ternary
3. only default header set — common case, hits the fallback at `index.ts:1337`

File: packages/layout-engine/layout-engine/src/index.ts [hunk 3]
```diff
@@ -1284,11 +1294,10 @@ export function layoutDocument(blocks: FlowBlock[], measures: Measure[], options
...
         const titlePgEnabled = sectionMetadata?.titlePg ?? false;
-        // TODO: Support alternateHeaders (odd/even) when needed
-        const alternateHeaders = false;
+        const alternateHeaders = options.alternateHeaders ?? false;
 
         // Determine which header/footer variant applies to this page
-        const variantType = getVariantTypeForPage(sectionPageNumber, titlePgEnabled, alternateHeaders);
+        const variantType = getVariantTypeForPage(sectionPageNumber, newPageNumber, titlePgEnabled, alternateHeaders);
 
         // Resolve header/footer refs for margin calculation using OOXML inheritance model.
```
Comment (caio-pizzol) [inline packages/layout-engine/layout-engine/src/index.ts:1325]: checked ECMA §17.10.5 — your code is right. cross-section inheritance is the spec behavior.

FYI only: renderer (`headerFooterUtils.ts:364`) doesn't do the same inheritance — returns `default` where your layout returns `even`. pre-PR this didn't matter (alternateHeaders was always false). now it does. worth a follow-up.

**Review by caio-pizzol — COMMENTED** [review]
@gpardhivvarma nice fix. tested against a few Word docs — the core change works.

small cleanup: `alternateHeaders` gets added to the options object in semantic flow too, but the type only allows it in paginated. inline below.

tests — all 4 are single-section, so the actual fix isn't exercised (old buggy code would pass them). left drop-in suggestions inline.

FYI, not a blocker: your PR surfaces a renderer bug. when section 2 doesn't define an even header, your layout correctly inherits from section 1 (per ECMA §17.10.5), but the renderer in `headerFooterUtils.ts` paints the default. so the page reserves space for the tall header but paints the short one. follow-up issue.

[fixture-c-section2-starts-even.docx](https://github.com/user-attachments/files/26908555/fixture-c-section2-starts-even.docx)

**Comment by gpardhivvarma** [comment]
@caio-pizzol thanks for the thorough review — all three items addressed:

1. **Type guard** — `alternateHeaders` now only added in paginated flow via `isSemanticFlow` check
2. **Multi-section test** — section 2 starts on page 4 (even by document number, odd by section number) — directly exercises the `documentPageNumber` fix
3. **Footer test** — even/odd footer heights with `alternateHeaders`
4. **Default-only fallback** — only `default` header defined, verifies the common case

Re: the renderer mismatch at `headerFooterUtils.ts:364` — noted, happy to do a follow-up PR for that.

**Review by caio-pizzol — COMMENTED** [review]
pushed two more commits:

- rewrote the footer and fallback tests — they weren't catching bugs.
- moved the odd/even flag next to the other page-level options.
- added the missing import, wiring, and browser tests.

opened a real Word doc to double-check: every page gets the right header, and saving it back out keeps everything intact. uploaded the doc to our corpus.

renderer follow-up from round 1 still open, not blocking.

lgtm.
